core/src/main/java/me/aurium/beetle/api/task/TaskFutureException.java

```diff
@@ -0,0 +1,9 @@
+package me.aurium.beetle.api.task;
+
+public class TaskFutureException extends RuntimeException{
```

Comment:
The JDK uses CompletionException to represent an exception thrown from an asynchronous computation, and I suggest you do too.

Considering the comments above you may not need this exception at all either.